src/libraries/Microsoft.PowerFx.Connectors/Public/CdpRecordType.cs

```diff
@@ -77,12 +82,15 @@ public override bool TryGetFieldType(string fieldName, out FormulaType type)
             }
         }
 
-#pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
         public override bool Equals(object other)
         {
-            throw new NotImplementedException();
+            if (other is not CdpRecordType crt)
+            {
+                return false;
+            }
+
+            return ConnectorType.FormulaType.Equals(crt.ConnectorType.FormulaType);
```

Comment:
>FormulaType [](http://example.com/codeflow?start=33&length=11)

ConnectotYpe.FormulaType is a FormulaType.
This method is an override of FormulaTYpe.Equals. 
How do we know this won't call itself and stackoverflow?

Reply:
ConnectorType.FormulaType won't include a CdpRecordType if external tables are involved